### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+        MovieSession movieSession = new MovieSession();
+        movieSession.setCinemaHall(cinemaHall);
+        movieSession.setMovie(movie);
+        movieSession.setShowTime(LocalDateTime.of(LocalDate.now(), LocalTime.of(20, 00)));
+        MovieSessionService movieSessionService = (MovieSessionService) injector
+                .getInstance(MovieSessionService.class);
+        movieSessionService.add(movieSession);
+        System.out.println(movieSessionService
+                .findAvailableSessions(movie.getId(), LocalDate.now()));
     }
```
Comment (Victoria-Moiseienko) [inline src/main/java/com/dev/cinema/Main.java:59]: You had better add several movieSessions for testing

**Review by Victoria-Moiseienko — COMMENTED** [review]
Good job!

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("from MovieSession ms where "
+                    + "ms.id = :movieId and ms.showTime between :start and :end");
+            query.setParameter("movieId", movieId);
+            query.setParameter("start", date.atTime(LocalTime.MIN));
+            query.setParameter("end", date.atTime(LocalTime.MAX));
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get movie sessions", e);
```

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/MovieSession.java]: what about access modifiers?

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "cinema_hall")
+public class CinemaHall {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/CinemaHall.java]: what about access modifiers?

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 3 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/model/MovieSession.java]: ooops)))

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/MovieSession.java:19]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save (same goes for other methods with input parameters, where we throw exception)

**Review by RomanDubovskyi — COMMENTED** [review]
make exception messages more informative, rest is good

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 5 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/model/MovieSession.java:19]: в идеале нужно указать fetch type lazy и потом, когда я буду писать HQL запрос, использовать join fetch ?

Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/model/MovieSession.java:19]: it's optional for this task, just keep that in mind if you need to improve performance that might be useful

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**